### PR TITLE
docs: meta configurations

### PR DESCRIPTION
**Pull request by soyeric128**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR updates Meta configurations available in [databend-meta.toml](https://github.com/datafuselabs/databend/blob/main/scripts/distribution/configs/databend-meta.toml). 

@PsiACE 

- Closes #issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/12682)

### CODE REVIEW

**Review by PsiACE — APPROVED** [review]

File: docs/doc/10-deploy/10-node-config/01-metasrv-config.md [hunk 1]
```diff
@@ -2,113 +2,22 @@
...
+| [raft_config] id                  | Unique identifier for the Raft configuration.                                                                           |
+| [raft_config] raft_dir            | Directory where Raft data is stored.                                                                                    |
+| [raft_config] raft_api_port       | Port for the Raft API of Databend.                                                                                      |
+| [raft_config] raft_listen_host    | IP address for Raft to listen on.                                                                                       |
+| [raft_config] raft_advertise_host | IP address used for advertising the Raft API.                                                                           |
+| single                          | Boolean indicating whether Databend should run in single-node cluster mode (true or false).                             |
+| join                            | List of addresses (<raft_advertise_host>:<raft_api_port>) of nodes in an existing cluster that a new node is joined to. |
```
Comment (drmingdrmer) [inline docs/doc/10-deploy/10-node-config/01-metasrv-config.md:23]: single and join are both `raft_config` too.

File: docs/doc/10-deploy/10-node-config/01-metasrv-config.md [hunk 2]
```diff
@@ -2,113 +2,22 @@
...
+
+- Some parameters listed in the table may not be present in [databend-meta.toml](https://github.com/datafuselabs/databend/blob/main/scripts/distribution/configs/databend-meta.toml). If you require these parameters, you can manually add them to the file.
+
+- You can find [sample configuration files](https://github.com/datafuselabs/databend/tree/main/scripts/ci/deploy/config) on GitHub that set up Databend for various deployment environments. These files were created for internal testing ONLY. Please do NOT modify them for your own purposes. But if you have a similar deployment, it is a good idea to reference them when editing your own configuration files.
+
+| Parameter                       | Description                                                                                                             |
+|---------------------------------|-------------------------------------------------------------------------------------------------------------------------|
+| log_dir                         | Directory where Databend stores its log files.                                                                          |
```
Comment (drmingdrmer) [inline docs/doc/10-deploy/10-node-config/01-metasrv-config.md:13]: `log_level` and `log_dir` are obsolete and should be removed from doc or marked as `deprecated`.

The replacement is `log: LogConfig`, which provide rich log config such as separate stderr log config and file log config. This doc should explain or add link to `log: LogConfig` here.

**Review by drmingdrmer — COMMENTED** [review]
Thank you for the improvements made to the meta config doc.

I would recommend to keep the basic config snippet here to provide immediate understanding of what the configuration looks like.

**Comment by drmingdrmer** [comment]
BTW, there is a sample section for the lacking `log`:

### Section `[log]`

Databend-meta provides two log config:

- `[log.file]` specifies the on disk file log config.
- `[log.stderr]` specifies the on disk file log config.


```toml
[log.file]
on = true
level = "DEBUG"
dir = "./.databend/logs"
format = "json"

[log.stderr]
on = true
level = "DEBUG"
format = "text"
```

### Section `[log.file]`

- `dir`: Directory where Databend stores its log files.

- `on`: is a bool indicates whether file based log is enabled.
- `level`: speicifies the log level.
- `format`: specifies the log format to be either `text` or `json`.

### Section `[log.stderr]`

`stderr` log has the same config entries except `dir`.

- `on`: is a bool indicates whether stderr log is enabled.
- `level`: speicifies the log level.
- `format`: specifies the log format to be either `text` or `json`.

**Pull request merged**